src/k8s/pkg/utils/sys.go

```diff
@@ -3,22 +3,38 @@ package utils
 import (
 	"context"
 	"fmt"
+	"io"
 	"os"
 	"os/exec"
+	"strings"
 )
 
 // runCommand executes a command with a given context.
 // runCommand returns nil if the command completes successfully and the exit code is 0.
 func RunCommand(ctx context.Context, command ...string) error {
+	return RunCommandWithCustomWriter(ctx, os.Stdout, os.Stderr, command...)
+}
+
+func RunCommandWithCustomWriter(ctx context.Context, out io.Writer, err io.Writer, command ...string) error {
 	var args []string
 	if len(command) > 1 {
 		args = command[1:]
 	}
 	cmd := exec.CommandContext(ctx, command[0], args...)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+	cmd.Stdout = out
+	cmd.Stderr = err
 	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("command %v failed with exit code %d: %w", command, cmd.ProcessState.ExitCode(), err)
 	}
 	return nil
 }
+
+func RunsWithRootPrivilege(ctx context.Context) (bool, error) {
+	out := strings.Builder{}
+	errOut := strings.Builder{}
+	err := RunCommandWithCustomWriter(ctx, &out, &errOut, "id", "-u")
+	if err != nil {
+		return false, fmt.Errorf("failed to check for privilege: %w, %s", err, errOut.String())
+	}
+	return strings.TrimSpace(out.String()) == "0", nil
+}
```

Comment:
use https://pkg.go.dev/os/user#Current instead

Reply:
this means we can also revert the `RunCommand` changes?

Reply:
What do you mean? This only checks if the k8s command was executed as root and returns a proper error message if not.

Reply:
I mean we don't need the `RunCommandWithCustomWriter` (not yet)